### PR TITLE
feat: update to new operator-sdk using Reconsiler

### PR DESCRIPTION
**Pull request by aslakknutsen**
* clean up deps
* add deploy-operator, deploy-example, undeploy-operator, undeploy-example make targets

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,4 +73,36 @@ docker-build: ## Builds the docker image
...
+	@echo "UnDeploying operator to $(OPERATOR_NAMEPACE)"
+	oc delete -f deploy/operator.yaml -n $(OPERATOR_NAMEPACE)
+	oc delete -f deploy/role_binding.yaml -n $(OPERATOR_NAMEPACE)
+	oc delete -f deploy/role.yaml -n $(OPERATOR_NAMEPACE)
+	oc delete -f deploy/service_account.yaml -n $(OPERATOR_NAMEPACE)
+	oc delete -f deploy/crds/istio_v1alpha1_session_crd.yaml -n $(OPERATOR_NAMEPACE)
+
+.PHONY:
+deploy-example:
+	@echo "Deploying operator to $(EXAMPLE_NAMEPACE)"
```
Comment (bartoszmajsak) [inline Makefile]: `EXAMPLE_NAMEPACE` -> `EXAMPLE_NAMESPACE`

File: cmd/ike/cmd/root.go [hunk 2]
```diff
@@ -28,18 +38,65 @@ Duis nec interdum velit, id consectetur erat. In tempor tempor turpis vel rhoncu
...
 	if err != nil {
-		logrus.Fatalf("Failed to get watch namespace: %v", err)
+		log.Error(err, "Failed to get watch namespace")
+		os.Exit(1)
+	}
+
+	// Get a config to talk to the apiserver
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "")
```
Comment (bartoszmajsak) [inline cmd/ike/cmd/root.go]: Are you moving away from `logrus` and switching to `log`?

Comment (aslakknutsen) [reply cmd/ike/cmd/root.go]: logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"

https://github.com/aslakknutsen/istio-workspace/pull/6/files#diff-b9de94200d7321a9f4eb6309638db08cR13

File: Makefile (hunk 1 above)
Comment (aslakknutsen) [reply Makefile]: https://github.com/aslakknutsen/istio-workspace/pull/6/commits/d88ab7bc81f251a46d33b7832862e60b172508f5

**Pull request merged**